astroquery/nist/tests/test_nist_remote.py

```diff
@@ -30,3 +30,23 @@ def test_unescape_html(self):
         # raw HTML code equivalents during parsing
         response = nist.core.Nist._parse_result(response)
         assert any('†' in s for s in response['Ei           Ek'])
+
+    def test_query_limits(self):
+        result = nist.core.Nist.query(4101 * u.AA, 4103 * u.AA)
+        # check that min, max wavelengths are appropriately set
+        assert result['Ritz'].min() >= 4101
+        assert result['Ritz'].max() <= 4103
+
+        # check that the units are respected
+        result = nist.core.Nist.query(410.1 * u.nm, 410.3 * u.nm)
+        assert result['Ritz'].min() >= 410.1
+        assert result['Ritz'].max() <= 410.3
+
+        result = nist.core.Nist.query(0.4101 * u.um, 0.4103 * u.um)
+        assert result['Ritz'].min() >= 0.4101
+        assert result['Ritz'].max() <= 0.4103
+
+        # check that non-supported units default to angstroms
+        result = nist.core.Nist.query(4101 * 1e-10 * u.m, 4103 * 1e-10 * u.m)
+        assert result['Ritz'].min() >= 4101
+        assert result['Ritz'].max() <= 4103
```

Comment:
Wow, this is totally unexpected. Would you mention it in the narrative docs?

Or, alternative we should instead return a QTable with units properly set on the column.

Reply:
QTable seems the better solution; I think we should punt to the next PR though.

Reply:
https://github.com/astropy/astroquery/issues/2920